plugins/badges-backend/src/service/router.ts

```diff
@@ -85,11 +87,10 @@ export async function createRouter(
     '/entity/:namespace/:kind/:name/badge/:badgeId',
     async (req, res) => {
       const { namespace, kind, name, badgeId } = req.params;
+      const token = await tokenManager.getToken();
       const entity = await catalog.getEntityByRef(
         { namespace, kind, name },
-        {
-          token: getBearerToken(req.headers.authorization),
-        },
+        token,
```

Comment:
This is actually quite problematic, because it bypasses user permissions. There's some more info in this problem in #14659, along with a few recommended paths forward.